### DISCUSSION TITLE
Is there a limit to how much data can be passed along with Link state?

### DISCUSSION THREAD

**Discussion by cem-al** [remix-run/remix#9411] [Q&A] [answered]
I'm curious about the limits or best practices when it comes to the amount of data that can be passed through the state prop. 

I need to pass an array of IDs from one page to another. The array structure looks something like this: `['nanoid1', 'nanoid2', 'nanoid3', ...]`, where each ID is a string of 7 characters.

Is there a practical limit to the number of IDs that can be passed through the state prop? For example, could I pass 500 IDs? 5000 IDs? At what point would it be considered too much?

Are there any performance implications?

**Comment by kiliman** [comment, marked as answer]
https://developer.mozilla.org/en-US/docs/Web/API/History/pushState#state

> Note: Some browsers save state objects to the user's disk so they can be restored after the user restarts the browser, and impose a size limit on the serialized representation of a state object, and will throw an exception if you pass a state object whose serialized representation is larger than that size limit. So in cases where you want to ensure you have more space than what some browsers might impose, you're encouraged to use [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage) and/or [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).
